File: src/compiler/typescript_to_rust/generators/ic_object/functions/stable64_size.ts
import { Rust } from '../../../../../types';

export function generateIcObjectFunctionStable64Size(): Rust {
    return /* rust */ `
        fn _azle_ic_stable64_size(
            _this: &boa_engine::JsValue,
            _aargs: &[boa_engine::JsValue],
            _context: &mut boa_engine::Context
        ) -> boa_engine::JsResult<boa_engine::JsValue> {
            Ok(ic_cdk::api::stable::stable64_size().azle_into_js_value(_context))
        }
    `;
}

File: src/compiler/typescript_to_rust/azle_generate/src/body/ic_object/functions/stable64_size.rs
pub fn generate() -> proc_macro2::TokenStream {
    quote::quote! {
        fn stable64_size(
            _this: &boa_engine::JsValue,
            _aargs: &[boa_engine::JsValue],
            context: &mut boa_engine::Context
        ) -> boa_engine::JsResult<boa_engine::JsValue> {
            Ok(ic_cdk::api::stable::stable64_size().try_into_vm_value(context).unwrap())
        }
    }
}
